script/download_MNIST.py

```diff
@@ -1,34 +1,64 @@
-__author__ = "Xinqiang Ding <xqding@umich.edu>"
+__author__ = "Keenan Manpearl"
+__date__ = "2023/1/24"
+
+"""
+original code by Xinqiang Ding <xqding@umich.edu>
+download MNIST data from Yann Lecun's website 
+[LeCun et al., 1998a]
+    Y. LeCun, L. Bottou, Y. Bengio, and P. Haffner. 
+        "Gradient-based learning applied to document recognition." 
+        Proceedings of the IEEE, 86(11):2278-2324, November 1998.
+"""
 
 import numpy as np
 import urllib3
 import gzip
-import subprocess
 import pickle
 
-## download train labels
+# download train labels
+# download train labels
 print("Downloading train-labels-idx1-ubyte ......")
 http = urllib3.PoolManager()
 r = http.request('GET', 'http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz')
 data = gzip.decompress(r.data)
+# number of images is 4-8
+# number of images is 4-8
 num = int.from_bytes(data[4:8], 'big')
+# images start at 8
```

Comment:
hmm, it looks like many of the comments below are redundant. I bet this is some sort of glitch in the IDE you're using maybe?

Please remove all redundant comments